### DISCUSSION TITLE
Generate an arbitrary url path

### DISCUSSION THREAD

**Discussion by jasikpark** [dubzzz/fast-check#3170] [Q&A] [answered]
`fc.webSegment` and `fc.webUrl` exists, but idk how to craft my own `fc.webPath`

**Comment by dubzzz** [comment, marked as answer]
Are you looking for https://github.com/dubzzz/fast-check/blob/main/packages/fast-check/src/arbitrary/webUrl.ts#L73? (I mean `pathArb`)? For the moment, we have not exposed it, but we should probably if it can be needed on its own 🤔

For now, as it has not been exposed yet, I'd recommend you to take inspiration from it. But definitely if useful, we should expose it to users so that they can use it directly.

**Reply by dubzzz** [reply]
If it's the one you were looking for, then I suggest we expose it.

**Reply by jasikpark** [reply]
yep that's what i'm looking for

**Comment by jasikpark** [comment]
Currently I'm using:

```ts
 fc.assert(
      fc.property(
        fc
          .array(
            fc.webSegment().filter((s) => !!s.length),
            { minLength: 1 }
          )
          .map((segments) => {
            return `/${segments.join('/')}`;
          }),
        fc.context(),
        (input, ctx) => {
          ctx.log(`input: ${input}, output: ${validateStatsPath(input)}`);
          expect(validateStatsPath(input)).toBe(true);
        }
      )
    );
```

as my setup, which is also what prompted #3179 since I see no way to ensure that an output of `fc.webSegment()` isn't an empty string other than a filter
